src/components/CommentAdder.jsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react'
+import * as api from '../utils/api'
+import './commentAdder.css'
+
+const CommentAdder = ({article_id, setComments}) => {
+    const [newComment, setNewComment] = useState("")
+    const [isSuccessful, setIsSuccessful] = useState(false)
+    const [error, setError] = useState(null)
+
+    const changeHandler = (e) => {
+        setNewComment(e.target.value)
+    }
+
+    const submitHandler = (e) => {
+        e.preventDefault();
+        setIsSuccessful(true)
+        api.postComments(article_id, newComment).then((addedComment) => {
+            setComments((currComments) => {
+                return [...currComments, addedComment]
+            })
+            setIsSuccessful(false)
+        }).catch((err) => {
+            setIsSuccessful(false)
+            setError("Something's gone worng, please refresh the page")
+        })
+        setNewComment("")
+    }
+    
+    if(isSuccessful) <p>Posting...</p>
+    if(error) return error;
+
+  return (
+    <form onSubmit={submitHandler} className='container form__container'>
+        <p></p>
+        <div id='comment-form'>
+        {/* User avatar goes here */}
+        <input id='add-comment' value={newComment} onChange={changeHandler} type='text' placeholder='Add a comment...'></input>
+        <button id='post-comment-btn'>Post</button>
```

Comment:
I would love to see you disable this button if the post is pending (the best way to demo this is to slow your internet connection down (throttle it) and try and post a comment whilst pressing the submit button loads of times). Ideally the user shouldn't be able to click the button at all until the post has succeeded (or failed) and the form has reset. At the moment when I try it I crash the whole thing